### DISCUSSION TITLE
Automatic Disk Health / Bad Block / Failure Checking?

### DISCUSSION THREAD

**Discussion by modem-19** [amidaware/tacticalrmm#2050] [General]
Good afternoon!  I am wondering how it's possible to have automatic disk health checking in TacticalRMM so that I can get email alerts for this.  Currently I have a PC I'm testing tRMM on with software RAID between two 1TB Drives.  This RAID is in a degraded state/failing but tRMM has not indicated this with any checks.

I would like to know if there is the possibility of tRMM sending out alerts for anything with hard drives (degraded raid, bad blocks, failures, etc) and whether this is a custom script that needs written, or if there is built in checks in tRMM that just need enabled.

Thanks!

**Comment by wh1te909** [comment]
try the built in community script check named "Disk Drive Volume Health Check (using Event Viewer errors)": https://github.com/amidaware/community-scripts/blob/main/scripts/Win_Disk_Volume_Status.ps1

can always modify it to your needs
